quickwit/quickwit-metastore/src/metastore/file_backed_metastore/file_backed_index.rs

```diff
@@ -489,3 +471,207 @@ impl Debug for Stamper {
         fmt.debug_struct("Stamper").field("stamp", &self.0).finish()
     }
 }
+
+fn split_query_predicate(split: &&Split, query: &ListSplitsQuery<'_>) -> bool {
+    if !split_tag_filter(split, query.tags.as_ref()) {
+        return false;
+    }
+
+    if !query.split_states.is_empty() && !query.split_states.contains(&split.split_state) {
+        return false;
+    }
+
+    if !query
+        .delete_opstamp
+        .contains(&split.split_metadata.delete_opstamp)
+    {
+        return false;
+    }
+
+    if !query.update_timestamp.contains(&split.update_timestamp) {
+        return false;
+    }
+
+    if !query
+        .create_timestamp
+        .contains(&split.split_metadata.create_timestamp)
+    {
+        return false;
+    }
+
+    if !query.time_range.is_unbounded() {
+        let is_in_range = split
+            .split_metadata
+            .time_range
+            .as_ref()
+            .map(|range| {
+                let start_check = match &query.time_range.start {
```

Comment:
We can make this more readable by adding an `overlaps_with` on `FilterRange` and then we can write:
```rust
if query.time_range.is_unbounded() {
    return true;
}
let Some(split_time_range) = &split.split_metadata.time_range else {
    return true;
}
query.time_range.overlaps_with(split_time_range)
```
That way, we can also easily write a unit test on `overlaps_with` because it's so easy to get wrong.

Reply:
Have implemented this and added unit tests.